[10.0][FIX] shopinvader: increase protection of the cart entry points

lmignon opened the pull request:
if trying to manipulate a cart that is not current anymore, ignore it, create a new cart; do not fail if removing an item that is not in cart anymore; create a new cart if adding an item when the current cart is closed

partial implementation of #373

lmignon committed:
[FIX] shopinvader: increase protection of the cart entry points
if trying to manipulate a cart that is not current anymore, ignore it, create a new cart; do not fail if removing an item that is not in cart anymore; create a new cart if adding an item when the current cart is closed

simahawk commented on shopinvader/services/cart.py:
```diff
@@ -241,22 +264,21 @@ def _get(self):
-        ]
         cart = self.env["sale.order"].browse()
         if self.cart_id:
+            # here we take advantage of the cache. If the card has been
```
*/card/cart ?

simahawk reviewed (approved):
LG

rousseldenis reviewed (approved):
Code review. Makes sense.

lmignon committed:
[FIX] shopinvader: Use the right pricelist on the cart
When creating an empty cart, takes the pricelist from the backend if one is defined.
We must avoid to put the field 'pircelist_id' into the dict used to create the SO if the value is False otherwise, the default pricelist will not be computed by odoo
Moreover, when the addon shopinvader_sale_profile is installed, it's not allowed to set a pricelit on the backend and the pricelist is computed from the sale profile.

lmignon commented:
https://github.com/shopinvader/odoo-shopinvader/pull/361 introduces some regressions with the pricelists
